### PR TITLE
Adding new test for USM

### PR DESCRIPTION
**Pull request by vasilytric**
Provide verification for functions/APIs from Unified Shared Memory (USM) fill, memcpy and memset

### CODE REVIEW

File: tests/usm/usm_fill_memset_memcpy.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      constexpr auto count{2};
+      constexpr auto size{sizeof(int) * count};
+      constexpr auto value_for_filling{1};
+      constexpr auto value_for_first_element_overwriting{10};
+
+      auto deleter = [&q](int *data) { sycl::free(data, q.get_context()); };
+      std::unique_ptr<int, decltype(deleter)> src(
+          sycl::malloc_shared<int>(size, q.get_device(), q.get_context()),
+          deleter);
+      std::unique_ptr<int, decltype(deleter)> output(
```
Comment (keryell) [inline tests/usm/usm_fill_memset_memcpy.cpp:48]:
```suggestion
      std::unique_ptr src(
          sycl::malloc_shared<int>(size, q.get_device(), q.get_context()),
          deleter);
      std::unique_ptr output(
```
thanks to C++17 CTAD

Comment (yuriykoch) [reply tests/usm/usm_fill_memset_memcpy.cpp:48]: @keryell How will it know that's an `unique_ptr<int>` or `unique_ptr<int[]>` though?

Comment (vasilytric) [reply tests/usm/usm_fill_memset_memcpy.cpp:48]: @keryell  I've tried to apply your suggestion but got a compilation error: "usm_fill_memset_memcpy.cpp(45,23): error: no viable constructor or deduction guide for deduction of template arguments of 'unique_ptr'
      std::unique_ptr src("

File: tests/usm/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+file(GLOB test_cases_list *.cpp)
```
Comment (ProGTX) [inline tests/usm/CMakeLists.txt:1]: It's generally bad CMake practice to glob for source files, it's better to list them explicitly in order to be able to properly track file dependencies.

File: tests/usm/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+
+add_cts_test(${test_cases_list})
```
Comment (ProGTX) [inline tests/usm/CMakeLists.txt:3]: USM is a new test category, could you please add it to the standard filter as well?

**Review by ProGTX — APPROVED** [review]

File: tests/usm/CMakeLists.txt (hunk 2 above)
Comment (vasilytric) [reply tests/usm/CMakeLists.txt:1]: This decision implemented across all repository. Fixing this task out of scope this pull request

File: tests/usm/CMakeLists.txt (hunk 3 above)
Comment (vasilytric) [reply tests/usm/CMakeLists.txt:3]: i've applied this comment in https://github.com/KhronosGroup/SYCL-CTS/pull/127/commits/83da3e852fb18ead362208f0f9a64915b65c304c

File: tests/usm/usm_fill_memset_memcpy.cpp (hunk 1 above)
Comment (vasilytric) [reply tests/usm/usm_fill_memset_memcpy.cpp:48]: Hi Ronan. I've tried to apply your suggestion but got a compilation error. I have no other ideas for changing this part of code.

File: tests/usm/CMakeLists.txt (hunk 2 above)
Comment (ProGTX) [reply tests/usm/CMakeLists.txt:1]: We could start doing it properly just here for now, but I'm happy either way

File: tests/usm/usm_fill_memset_memcpy.cpp (hunk 1 above)
Comment (keryell) [reply tests/usm/usm_fill_memset_memcpy.cpp:48]: Curious. It looks like this is by design according to https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr : "The program is ill-formed if this constructor is selected by class template argument deduction."
This does not work even for simple cases: https://godbolt.org/z/9GeqbTd8h
So, keep it like you did. Thanks for having tried.

Comment (vasilytric) [reply tests/usm/usm_fill_memset_memcpy.cpp:48]: @keryell  Do I understand correctly that you approve this PR?

**Review by keryell — APPROVED** [review]

File: tests/usm/usm_fill_memset_memcpy.cpp (hunk 1 above)
Comment (keryell) [reply tests/usm/usm_fill_memset_memcpy.cpp:48]: Oh yes! Thank you for reminding this. Sorry for the distraction...

**Pull request merged**